range matching fails

When I set `engines.node: ">=6"` in the package.json, I expect dynamic imports to evoke an error (see [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#browser_compatibility)). They don't. When I set an upper boundary, like `>=6 <=12.16`, it _will_ throw an error.

I recreated the error here, using this project's code.
```js
import semver from 'semver';

const supported = new semver.Range('>=12.17 <13 || >=13.2');

const versionFailing = new semver.Range('>=6');
const versionPassing = new semver.Range('>=6 <=12.16');

console.log(versionFailing.intersects(supported)); // In the code, this means the rule is supported
// => true, because >=6 *includes* a node version which supports it.
```

Possible cause:
The code only checks that _some_ of the given versions support the code. ("intersects")
https://github.com/weiran-zsd/eslint-plugin-node/blob/592869528b1e6e3278603ea00696000019c22114/lib/rules/no-unsupported-features/es-syntax.js#L461

So, maybe I'm putting this together wrong, but here's what I've got:

https://gist.github.com/nazrhyn/88026aa52559750ceb2d95ea8c8ccbeb

Here's the output I get from ESLint:

```
$ npx eslint test.js

/home/nazrhyn/code/eslint-plugin-node-test/test.js
  2:9  error  'import()' expressions are not supported yet  node/no-unsupported-features/es-syntax

✖ 1 problem (1 error, 0 warnings)
```

@aladdin-add @binyamin Could you guys fix whatever I'm doing wrong?

--- a/lib/rules/no-unsupported-features/es-syntax.js
+++ b/lib/rules/no-unsupported-features/es-syntax.js
@@ -7,6 +7,7 @@
 const { rules: esRules } = require("eslint-plugin-es")
 const { getInnermostScope } = require("eslint-utils")
 const { Range } = require("semver")
+const rangeSubset = require("semver/ranges/subset")
 const getConfiguredNodeVersion = require("../../util/get-configured-node-version")
 const getSemverRange = require("../../util/get-semver-range")
 const mergeVisitorsInPlace = require("../../util/merge-visitors-in-place")
@@ -457,11 +458,11 @@ function defineVisitor(context, options) {
             return true
         }
 
-        if (aCase.supported instanceof Range) {
-            return !options.version.intersects(aCase.supported)
-        }
-
-        return options.version.intersects(getSemverRange(`<${aCase.supported}`))
+        const supported =
+            typeof aCase.supported === "string"
+                ? getSemverRange(`>=${aCase.supported}`)
+                : aCase.supported
+        return !rangeSubset(options.version, supported)
     }
 
     /**

--- a/tests/lib/rules/no-unsupported-features/es-syntax.js
+++ b/tests/lib/rules/no-unsupported-features/es-syntax.js
@@ -2493,7 +2493,7 @@ ruleTester.run(
                 })),
             ],
             invalid: [
-                ...["12.16.0", "13.0.0", "13.1.0"].map(v => ({
+                ...["12.16.0", "13.0.0", "13.1.0", ">=8.0.0"].map(v => ({
                     code: "import(source)",
                     options: [{ version: v }],
                     errors: [

